.jenkins

```diff
@@ -145,12 +145,14 @@ pipeline {
                         OMP_MAX_ACTIVE_LEVELS = 3
                         OMP_PLACES = 'threads'
                         OMP_PROC_BIND = 'spread'
+                        NVCC_WRAPPER_DEFAULT_COMPILER = 'clang++'
```

Comment:
Note: this bug only occurs when using `clang`. `NVCC_WRAPPER_DEFAULT_COMPILER` defaults to `g++` which does NOT trigger the bug.

This SO post contains some hints about the difference in implementations: https://stackoverflow.com/a/70673863.

Reply:
I'm confused. `HIP-ROCm-5.2` doesn't use `nvcc_wrapper` anyway so adding this line should not have any effect.

Reply:
Ok, maybe I'm overdoing this. This was the case when using `nvcc_wrapper` and Cuda, let me double check HIP locally.

Reply:
In the meantime, the build seems to fail as expected:
```
15: terminate called after throwing an instance of 'std::runtime_error'
15:   what():  Kokkos::Impl::ParallelReduce< HIP > could not find a valid execution configuration.
```

Reply:
Line removed as it is not needed to reproduce the issue:
[rocm-rtld-deepbind.log](https://github.com/kokkos/kokkos/files/11897787/rocm-rtld-deepbind.log)